examples/misc/programs/misc/src/lib.rs

```diff
@@ -79,8 +87,33 @@ pub struct TestStateCpi<'info> {
     misc2_program: AccountInfo<'info>,
 }
 
+#[derive(Accounts)]
+pub struct TestAssociatedAccount<'info> {
+    #[account(associated = authority, with = state)]
+    my_account: ProgramAccount<'info, AssociatedData>,
+    // Must be a signer since it will pay for the creation of the associated
+    // token account.
+    #[account(signer)]
+    authority: AccountInfo<'info>,
+    // Associated accounts can identify other instances to be used as seeds for
+    // the program derived address.
+    state: ProgramState<'info, MyState>,
+    // `rent` and `system_program` are *required* by convention, since the
+    // accounts are needed when creating the associated program address within
+    // the program.
+    rent: Sysvar<'info, Rent>,
+    system_program: AccountInfo<'info>,
+}
+
 #[account]
 pub struct Data {
     udata: u128,
     idata: i128,
 }
+
+#[account]
+#[derive(Default)]
```

Comment:
One open question is whether we want to force `#[account]`s to implement `Default`. For now, we'll leave it out, although aesthetically unpleasant.

Reply:
This is solved by a new `#[associated]` attribute to replace account, in the associated account instance.